### PR TITLE
prov/efa: Fix a bug in rxr_pkt_handle_send_error

### PR DESCRIPTION
**Pull request by zhngaj**
When RNR happens, if the packet is associated with a send
operation, it's supposed to do one of the following two:
(1) write an error to the application, and release the packet
if the resource management is disabled;
(2) queue the packet, and send later.

We were missing a 'else', so we ended up doing both (1) and (2).

This patch fixes the bug and also does code refactoring to
make the logic easier to understand.

Signed-off-by: Jie Zhang <zhngaj@amazon.com>

### CODE REVIEW

File: prov/efa/src/rxr/rxr_pkt_cmd.c [hunk 1]
```diff
@@ -654,7 +654,7 @@ void rxr_pkt_handle_send_error(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entr
 			 */
 			rxr_cq_write_tx_error(ep, pkt_entry->x_entry, FI_ENORX, 0);
 			rxr_pkt_entry_release_tx(ep, pkt_entry);
-		} if (prov_errno == IBV_WC_RNR_RETRY_EXC_ERR) {
+		} else if (prov_errno == IBV_WC_RNR_RETRY_EXC_ERR) {
```
Comment (wzamazon) [inline prov/efa/src/rxr/rxr_pkt_cmd.c:657]: I think the code would be easier to understand that `prov_errno == IBV_WC_RNR_RETRY_EXC_ERR` has its own block.

Comment (zhngaj) [reply prov/efa/src/rxr/rxr_pkt_cmd.c]: Thanks. I updated in the new rev.

**Review by wzamazon — APPROVED** [review]

File: prov/efa/src/rxr/rxr_pkt_cmd.c [hunk 2]
```diff
@@ -646,26 +646,27 @@ void rxr_pkt_handle_send_error(struct rxr_ep *ep, struct rxr_pkt_entry *pkt_entr
...
+			if (ep->handle_resource_management == FI_RM_DISABLED) {
+				/*
+				 * Write an error to the application for RNR when
+				 * resource management is disabled.
+				 */
+				rxr_cq_write_tx_error(ep, pkt_entry->x_entry, FI_ENORX, 0);
+				rxr_pkt_entry_release_tx(ep, pkt_entry);
+			} else {
+				/*
+				 * This packet is assoiciated with a send operation,
```
Comment (rwespetal) [inline prov/efa/src/rxr/rxr_pkt_cmd.c:659]: Can you fix the typos in the comments while you're here?

**Review by rwespetal — APPROVED** [review]

File: prov/efa/src/rxr/rxr_pkt_cmd.c (hunk 2 above)
Comment (zhngaj) [reply prov/efa/src/rxr/rxr_pkt_cmd.c]: Sure. Pushed the fix.

**Pull request merged**